aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2334,7 +2383,205 @@ Tensor matrix_exp_backward(const Tensor& self, const Tensor& grad) {
+Tensor linalg_norm(const Tensor& X, const optional<Scalar>& opt_ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype) {
+  if (opt_dim.has_value()) {
+    TORCH_CHECK(opt_dim->size() == 1 || opt_dim ->size() == 2, "linalg.norm: If ",
+              "dim is specified, it must be of length 1 or 2. Got ", *opt_dim);
+  } else {
+    if (opt_ord.has_value()) {
+      TORCH_CHECK(X.dim() == 1 || X.dim() == 2, "linalg.norm: If ",
+                  "dim is not specified but ord is, the input must be 1D or 2D. Got ", X.dim(), "D.");
+    }
+  }
+
+  // If ord=None, we'll always use the 2-norm or frob norm (which are the same) so we go through
+  // vector_norm
+  if (opt_ord.has_value() &&
+       ((opt_dim.has_value() && opt_dim->size() == 2) ||
+        (!opt_dim.has_value() && X.dim() == 2))) {
+    using Int = IntArrayRef::value_type;
+    auto dim = opt_dim.has_value() ? opt_dim.value().vec() : std::vector<Int>{0, 1};
+    return at::linalg_matrix_norm(X, *opt_ord, dim, keepdim, opt_dtype);
+  } else {
+    auto scalar_ord = opt_ord.value_or(Scalar(2.));
+    return at::linalg_vector_norm(X, scalar_ord, opt_dim, keepdim, opt_dtype);
+  }
+}
+
+Tensor& linalg_norm_out(const Tensor& X, const optional<Scalar>& opt_ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype, Tensor& result) {
+  auto out = at::linalg_norm(X, opt_ord, opt_dim, keepdim, opt_dtype);
+  at::native::resize_output(result, out.sizes());
+  result.copy_(out);
+  return result;
+}
+
+// Frobenius and nuclear norms
+Tensor linalg_norm(const Tensor& X, c10::string_view ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype) {
+  if (opt_dim.has_value()) {
+    TORCH_CHECK(opt_dim->size() == 1 || opt_dim ->size() == 2, "linalg.norm: If ",
+              "dim is specified, it mut be of length 1 or 2. Got ", *opt_dim);
+  } else {
+    TORCH_CHECK(X.dim() == 1 || X.dim() == 2, "linalg.norm: If ",
+                "dim is not specified but ord is, the input must be 1D or 2D. Got ", X.dim(), "D.");
+  }
+  using Int = IntArrayRef::value_type;
+  auto dim = opt_dim.has_value() ? opt_dim.value().vec() : std::vector<Int>{0, 1};
+  return at::linalg_matrix_norm(X, ord, dim, keepdim, opt_dtype);
+}
+
+Tensor& linalg_norm_out(const Tensor& X, c10::string_view ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype, Tensor& result) {
+  auto out = at::linalg_norm(X, ord, opt_dim, keepdim, opt_dtype);
+  at::native::resize_output(result, out.sizes());
+  result.copy_(out);
+  return result;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//                         Frobenius Norm (deprecated)                          //
+////////////////////////////////////////////////////////////////////////////////
+
 Tensor frobenius_norm(const Tensor& self) {
+  TORCH_WARN_ONCE(
+    "torch.frobenius_norm is deprecated in favor of torch.linalg.matrix_norm(X) ",
```

Comment:
Let's separate further deprecation warnings into other PRs -- for this one we can probably not bother since it's an undocumented function (same for nuclear norm)